homeassistant/components/withings/common.py

```diff
@@ -255,32 +252,23 @@ def create_withings_data_manager(
     """Set up the sensor config entry."""
     entry_creds = entry.data.get(const.CREDENTIALS) or {}
     profile = entry.data[const.PROFILE]
-    credentials = withings.WithingsCredentials(
-        entry_creds.get("access_token"),
-        entry_creds.get("token_expiry"),
-        entry_creds.get("token_type"),
-        entry_creds.get("refresh_token"),
-        entry_creds.get("user_id"),
-        entry_creds.get("client_id"),
-        entry_creds.get("consumer_secret"),
+    credentials = Credentials(
```

Comment:
Note (unrelated to this specific line):

In #27727 I am introducing a generic OAuth2 config flow helper. It can work perfectly with `requests-oauthlib`, see the Somfy API for an example.

Reply:
This looks like it has the potential to reduce a lot of code. After it lands in dev, I'll implement it in another PR.